### PR TITLE
Fine-tuning Gemma 2 model using LoRA and Keras (Learning Index Path)

### PR DESCRIPTION
**Pull request by Foluwa**
This PR introduces fine-tuning of the Gemma 2 model using Low-Rank Adaptation (LoRA) to improve efficiency in adapting large model parameters. Utilizing the Learning Index Path (LPI) dataset and Keras, the fine-tuning aims to enhance model performance with minimal resource demand. This setup supports more scalable deployment and optimized inference times for downstream applications.

### CODE REVIEW

**Comment by neomatrix369** [comment]
@Foluwa please add a good title and description to this PR and also link it with one or more issues you are working on

File: app/llm-poc-variant-03/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Gemma Model Learning Index Path
+
+## Fine tune gemma model withcustom data
```
Comment (neomatrix369) [inline app/llm-poc-variant-03/README.md:3]: > gemma ==> Gemma

> withcustom ==> with custom

**Comment by Foluwa** [comment]
# Finetuning Gemma model on LPI Dataset


The initial approach was implemented in this [notebook](https://github.com/neomatrix369/learning-path-index/blob/lpi-gemma-model/app/llm-poc-variant-03/lpi_finetune_gemma.ipynb) but some errors were encountered (which are listed below) and I pivoted to this [notebook](https://github.com/neomatrix369/learning-path-index/blob/lpi-gemma-model/app/llm-poc-variant-03/lpi_keras_finetuning.ipynb) most of the implementation were from ["Fine-tuning Gemma 2 model with role-playing dataset"](https://medium.com/@gabi.preda/fine-tuning-gemma-2-model-with-role-playing-dataset-b8ec399a2e17) by Gabriel Preda

## Issues encountered

### 1.  Resource constraints (ensure you run on a GPU)
Screenshot 2024-10-26 at 5 15 34 PM
Screenshot 2024-10-26 at 4 17 20 PM
Screenshot 2024-10-26 at 4 17 33 PM


### 2.  Permission Issues
Screenshot 2024-10-26 at 4 19 27 PM
Screenshot 2024-10-26 at 4 18 05 PM


### 3. The evaluation metric does not support "multiclass-multioutput" format
Screenshot 2024-10-26 at 5 00 18 PM


### 4.  Unable to build the model
image1


### 5.  Unable to save the model weights
Screenshot 2024-10-26 at 5 11 14 PM


### 6. Model performance issue
Screenshot_2024-10-07_at_12 55 15_AM
